pkg/gitops/flux/installer.go

```diff
@@ -118,42 +104,56 @@ func (fi *Installer) Run(ctx context.Context) (string, error) {
 		return "", err
 	}
 
-	if fi.opts.WithHelm {
+	if api.IsEnabled(fi.opts.Operator.WithHelm) {
 		logger.Info("Waiting for Helm Operator to start")
-		if err := waitForHelmOpToStart(ctx, fi.opts.Namespace, fi.opts.Timeout, fi.k8sRestConfig, fi.k8sClientSet); err != nil {
+		if err := waitForHelmOpToStart(ctx, fi.opts.Operator.Namespace, fi.timeout, fi.k8sRestConfig, fi.k8sClientSet); err != nil {
 			return "", err
 		}
 		logger.Info("Helm Operator started successfully")
 		logger.Info("see https://docs.fluxcd.io/projects/helm-operator for details on how to use the Helm Operator")
 	}
 
 	logger.Info("Waiting for Flux to start")
-	fluxSSHKey, err := waitForFluxToStart(ctx, fi.opts.Namespace, fi.opts.Timeout, fi.k8sRestConfig, fi.k8sClientSet)
+	fluxSSHKey, err := waitForFluxToStart(ctx, fi.opts.Operator.Namespace, fi.timeout, fi.k8sRestConfig, fi.k8sClientSet)
 	if err != nil {
 		return "", err
 	}
 	logger.Info("Flux started successfully")
 	logger.Info("see https://docs.fluxcd.io/projects/flux for details on how to use Flux")
 
-	logger.Info("Committing and pushing manifests to %s", fi.opts.GitOptions.URL)
+	logger.Info("Committing and pushing manifests to %s", fi.opts.Repo.URL)
 	if err := fi.addFilesToRepo(ctx, cloneDir); err != nil {
 		return "", err
 	}
 	cleanCloneDir = true
 
 	logger.Info("Flux will only operate properly once it has write-access to the Git repository")
 	instruction := fmt.Sprintf("please configure %s so that the following Flux SSH public key has write access to it\n%s",
-		fi.opts.GitOptions.URL, fluxSSHKey.Key)
+		fi.opts.Repo.URL, fluxSSHKey.Key)
 	return instruction, nil
 }
 
+// IsFluxInstalled returns an error if Flux is not installed in the cluster. To determine that it looks for the flux
+// pod
+func (fi *Installer) IsFluxInstalled() (bool, error) {
+	deployments, err := fi.k8sClientSet.AppsV1().Deployments(fi.opts.Operator.Namespace).List(metav1.ListOptions{
+		FieldSelector: "metadata.name=flux"})
```

Comment:
If we just need to lookup by name (`metadata.name`) using `Get` would make the intent more clear.

```suggestion
	deployments, err := fi.k8sClientSet.AppsV1().Deployments(fi.opts.Operator.Namespace).Get("flux", metav1.GetOptions{})
```

But the error below would need to be checked against `StatusError` for 404 errors.

Reply:
Good point, thanks